R-package/tests/testthat/test_basic.R

```diff
@@ -2758,3 +2758,68 @@ for (x3_to_categorical in c(TRUE, FALSE)) {
     })
   }
 }
+
+test_that("lightgbm() accepts objective as function argument and under params", {
```

Comment:
Thanks for these tests!

Can you please add one more, `lightgbm() defaults to "regression" objective if objective not provided as function argument or under params`?

Since this PR is proposing adding a default for `objective`, I'd like to have a test explicitly capturing that behavior. User-facing changes to the package's behavior should be accompanied by tests, so that we can automatically catch the case where future proposed changes break existing behavior.

Reply:
Will add it, but the easy solution in such case would be to not have a default value so that the user would always need to supply it. That's, of course, a breaking change.